### PR TITLE
 Rpc cmd Hello; return JSON instead string in code

### PR DESCRIPTION
**Pull request by mikurys**
also added session_id and command_id to each rpc command

### CODE REVIEW

File: src/rpc/rpc.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@
 c_rpc_server::c_rpc_server(const unsigned short port)
 :
 	m_io_service(),
-	m_acceptor(m_io_service, boost::asio::ip::tcp::endpoint(boost::asio::ip::address_v4::from_string("127.0.0.1"), port)),
+	m_acceptor(m_io_service, boost::asio::ip::tcp::endpoint(boost::asio::ip::address_v4::any(), port)),
```
Comment (rfree-d) [inline src/rpc/rpc.cpp]: This needs to be a separate PR, that is marked as changing security;

For now do not do it, (except local tests, not committed here) - instead later add option to choose rpc bind adress.

File: src/rpc/rpc.cpp [hunk 2]
```diff
@@ -186,6 +188,12 @@ void c_rpc_server::c_session::set_iterator_in_session_list(std::list<c_session>:
 	m_iterator_in_session_list = it;
 }
 
+std::string c_rpc_server::c_session::get_command_id()
```
Comment (robertoleksy) [inline src/rpc/rpc.cpp]: never used?

File: src/rpc/rpc.hpp [hunk 3]
```diff
@@ -43,8 +45,9 @@ class c_rpc_server final {
 		std::thread m_thread;
 		Mutex m_session_vector_mutex;
 		std::list<c_session> m_session_list;
-		std::map<std::string, std::function<std::string(const std::string)>> m_rpc_functions_map;
+		std::map<std::string, std::function<nlohmann::json(const std::string)>> m_rpc_functions_map;
 		std::array<unsigned char, crypto_auth_hmacsha512_KEYBYTES> m_hmac_key; ///< for hmac authentication key, shold be loaded from conf file TODO
+		xint m_session_counter = 0;
```
Comment (robertoleksy) [inline src/rpc/rpc.hpp]: This variable can be use by multiple threads if we will add them. Should be GUARDED_BY(m_session_vector_mutex)

File: src/rpc/rpc.cpp (hunk 1 above)
Comment (mikurys) [reply src/rpc/rpc.cpp]: 🆗

File: src/rpc/rpc.cpp (hunk 2 above)
Comment (mikurys) [reply src/rpc/rpc.cpp]: it is used in line 215 in this file

File: src/rpc/rpc.hpp (hunk 3 above)
Comment (robertoleksy) [reply src/rpc/rpc.hpp]: I will fix mulithread bugs

**Comment by rfree-d** [comment]
UTack then:

Since io_service run is run just once, it is not causing actually a bug.

We can merge it, and improve it (prepare for more MT/many io_service-run) later.

Pointed out by @mikurys

**Pull request merged**